constexpr commit breaks clang++ compilation

Starting with commit e2268eadb69cd0fd2bf2e5ffcac5e30d03ebd41c , the generated vulkan.hpp fails compilation with clang++ (but not g++) with the following complaints. This is raising CI warnings on all the internal gitlab vulkan commits, since the hpp-compile test tests with both compilers, and hopefully can be addressed soon.

```
In file included from hpptest.cpp:1:
./vulkan/vulkan.hpp:2636:40: error: constexpr function's return type 'void' is
      not a literal type
      VULKAN_HPP_CONSTEXPR static void copy(T dst[N], std::array<T,N> co...
                                       ^
./vulkan/vulkan.hpp:2647:40: error: constexpr function's return type 'void' is
      not a literal type
      VULKAN_HPP_CONSTEXPR static void copy(T /*dst*/[N], std::array<T,N...
                                       ^
./vulkan/vulkan.hpp:25709:50: error: no member named 'copy' in
      'vk::ConstExpressionArrayCopy<float, 4, 4>'
        vk::ConstExpressionArrayCopy<float,4,4>::copy( color, color_ );
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
./vulkan/vulkan.hpp:35848:50: error: no member named 'copy' in
      'vk::ConstExpressionArrayCopy<float, 4, 4>'
        vk::ConstExpressionArrayCopy<float,4,4>::copy( blendConstants, b...
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
./vulkan/vulkan.hpp:36751:55: error: no member named 'copy' in
      'vk::ConstExpressionArrayCopy<vk::Offset3D, 2, 2>'
      vk::ConstExpressionArrayCopy<vk::Offset3D,2,2>::copy( dstOffsets, ...
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
./vulkan/vulkan.hpp:53126:93: error: no member named 'copy' in
      'vk::ConstExpressionArrayCopy<char, 256, 256>'
  ...vk::ConstExpressionArrayCopy<char,VK_MAX_DESCRIPTION_SIZE,VK_MAX_DESCRIPTION_SIZE>::c...
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
6 errors generated.
```

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -4924,7 +4924,7 @@ static const std::string constExpressionArrayCopy = R"(
   class ConstExpressionArrayCopy
   {
     public:
-      VULKAN_HPP_CONSTEXPR static void copy(T dst[N], std::array<T,N> const& src)
+      VULKAN_HPP_CONSTEXPR_14 static void copy(T dst[N], std::array<T,N> const& src)
       {
         dst[I-1] = src[I-1];
         ConstExpressionArrayCopy<T, N, I - 1>::copy(dst, src);
@@ -4935,7 +4935,7 @@ static const std::string constExpressionArrayCopy = R"(
   class ConstExpressionArrayCopy<T, N, 0>
   {
     public:
-      VULKAN_HPP_CONSTEXPR static void copy(T /*dst*/[N], std::array<T,N> const& /*src*/) {}
+      VULKAN_HPP_CONSTEXPR_14 static void copy(T /*dst*/[N], std::array<T,N> const& /*src*/) {}
   };
 )";
 

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -2660,7 +2660,7 @@ namespace VULKAN_HPP_NAMESPACE
   class ConstExpressionArrayCopy
   {
     public:
-      VULKAN_HPP_CONSTEXPR static void copy(T dst[N], std::array<T,N> const& src)
+      VULKAN_HPP_CONSTEXPR_14 static void copy(T dst[N], std::array<T,N> const& src)
       {
         dst[I-1] = src[I-1];
         ConstExpressionArrayCopy<T, N, I - 1>::copy(dst, src);
@@ -2671,7 +2671,7 @@ namespace VULKAN_HPP_NAMESPACE
   class ConstExpressionArrayCopy<T, N, 0>
   {
     public:
-      VULKAN_HPP_CONSTEXPR static void copy(T /*dst*/[N], std::array<T,N> const& /*src*/) {}
+      VULKAN_HPP_CONSTEXPR_14 static void copy(T /*dst*/[N], std::array<T,N> const& /*src*/) {}
   };
 
   using Bool32 = uint32_t;

